hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/RMProxy.java

```diff
@@ -116,6 +116,28 @@ protected static <T> T createRMProxy(final Configuration configuration,
     return newProxyInstance(conf, protocol, instance, retryPolicy);
   }
 
+  /**
+   * Currently, used by NodeManagers only.
+   *
+   * @param configuration configuration.
+   * @param protocol protocol.
+   * @param instance RMProxy instance.
+   * @return RMProxy.
+   * @param <T> Generic T.
+   * @throws IOException io error occur.
+   */
+  protected static <T> T createRMProxyFederation(final Configuration configuration,
+      final Class<T> protocol, RMProxy<T> instance) throws IOException {
+    YarnConfiguration yarnConf =
+        (configuration instanceof YarnConfiguration) ? (YarnConfiguration) configuration :
+        new YarnConfiguration(configuration);
+    if(isFederationNonHAEnabled(yarnConf)){
+      RetryPolicy retryPolicy = createRetryPolicy(yarnConf, true);
+      return newProxyInstance(yarnConf, protocol, instance, retryPolicy);
+    }
+    return createRMProxy(configuration, protocol, instance);
```

Comment:
Can we add a test?

Reply:
I will add unit tests as soon as possible.